base/challenge-skeleton/README.md

```diff
@@ -1,52 +1,61 @@
 
-## /config
+* `chal.conf`: Configure this file if you are deploying the challenge to the cluster, if it's publicly accessible, and/or if it has a healthcheck.
+* `pow.conf`: Contains the difficulty of the proof-of-work, 0 means disabled.
+* `advanced`: The Kubernetes config used to deploy the challenge. E.g. you can add a tmpfs here or change the port.
 
-The config directory holds a few configuration files:
+### /challenge
 
-* chal.conf: Configure if the challenge should be deployed to the cluster, if it's publicly accessible and if it has a healthcheck.
-* pow.conf: Contains the difficulty of the proof-of-work, 0 means disabled.
-* advanced: The kubernetes config used to deploy the challenge. E.g. you can add a tmpfs here or change the port.
+The `challenge` directory contains the challenge binary and anything else related to the challenge:
 
-## /challenge
+* `image/`: Dockerfile and files of the challenge image.
+* `Makefile`: Edit this file if you need to run code before building the Docker image, e.g. for building the challenge from source.
 
-The challenge binary and anything else challenge related goes in here.
+### /healthcheck
 
-* image/: Dockerfile and files of the challenge image.
-* Makefile: You can edit this file if you need to run code before building the docker image, e.g. for building the challenge from source.
+The `healthcheck` directory is optional. If you don't want to write a healthcheck, feel free to delete the directory. However, we strongly recommend that you implement a healthcheck :).
 
-## /healthcheck
+* `image/`: Dockerfile and files of the healthcheck.
+* `Makefile`: Edit this file if you need to build any custom files before running your healthcheck.
 
-The healthcheck directory is optional. If you don't want to write a healthcheck, feel free to delete it. You really should have healthchecks though :).
 
-* image/: Dockerfile and files of the healthcheck.
-* Makefile: same as for the challenge directory.
+#### Healthcheck
+
+Important details regarding healthchecks:
+
+* Run the healthcheck on a webserver on port 45281 that responds to `/healthz` requests.
+ * The healthcheck returns 200 if the challenge is healthy, otherwise an error (e.g. 400).
+ * You can find an example webserver in `healthcheck/image/healthz.py`.
+* The base image comes preloaded with pwntools.
+* If your exploit is written in Python, drop it in `healthcheck/image/doit.py`.
 
-# API contract
+## API contract
 
-Here are the requirements what this directory must look like to work with kCTF:
+Ensure your setup fulfills the following requirements to ensure it works with kCTF:
 
-* You can do pretty much whatever you want in the challenge dir but:
-  * You need to have a Makefile with the .gen/docker-image target that builds a docker image.
-  * We strongly recommend to use nsjail in all challenges.
-* Your challenge will receive connections on port 1337.
+* In your cmdline, call `kctf_setup` as the first command.
```

Comment:
Maybe we should add an example here. It's not clear to me what this means.
```Dockerfile
CMD kctf_setup && kctf_nsjail /home/user/chal
```

Reply:
or link to the Dockerfile?
We could also say "leave kctf_setup as the first command in the Dockerfile" or similar

Reply:
Yea let's do that :-)

Reply:
@goedi02 unlike the Makefile below, I think the Dockerfile would be clear (it's less magical than the Makefile below as it's not customized too much).

Reply:
ok, so we could write: "Ensure kctf_setup is used as the first command in your [Dockerfile](https://github.com/google/kctf/blob/master/base/challenge-skeleton/challenge/image/Dockerfile)"

Reply:
maybe people might get confused as a Dockerfile has many things (RUN, FROM, COPY, etc) and here we are talking about the "command" (CMD). Maybe we can explicitly call out "Ensure kctf_setup is used as the first command in your Dockerfile's CMD instruction?" or something like that.
https://docs.docker.com/engine/reference/builder/#cmd has some documentation about what CMD is

Reply:
Thank you for the clarification. We definitely need to be explicit about the CMD instruction, agreed! Changed sentence, including link to the Dockerfile.